jinahub/encoders/image/TimmImageEncoder/timm_encoder.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torchvision.transforms as T
+from jina import DocumentArray, Executor, requests
+from jina_commons.batching import get_docs_batch_generator
+from timm import create_model
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+
+
+class TimmImageEncoder(Executor):
+    """
+    :class:`TimmImageEncoder` encodes ``Document`` blobs of type `ndarray`
+    (`float32`) and shape `H x W x C` into `ndarray` of `D`.
+    Where `D` is the Dimension of the embedding.
+
+    If `use_default_preprocessing=False`, the expected input shape is
+    `C x H x W` with `float32` dtype.
+
+    Internally, :class:`TimmImageEncoder` wraps the pre-trained models from
+    `Timm library`.
+    https://rwightman.github.io/pytorch-image-models/
+
+    :param model_name: the name of the model. Models listed on:
+        https://rwightman.github.io/pytorch-image-models/models/
+    :param device: Which device the model runs on. Can be 'cpu' or 'cuda'.
+    :param default_traversal_paths: Used in the encode method an defines traversal on the received `DocumentArray`.
+    :param default_batch_size: Defines the batch size for inference on the loaded PyTorch model.
```

Comment:
Can you reword this to make it clearer they are defaults, and can be overwritten?